### PR TITLE
breaking: refactored module

### PR DESCRIPTION
**Pull request by thulasirajkomminar**
refactored module

### CODE REVIEW

File: iam.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module "job_execution_role" {
+  count                 = var.role_arn == null ? 1 : 0
+  source                = "github.com/schubergphilis/terraform-aws-mcaf-role?ref=v0.4.0"
+  name                  = "GlueExecution-${var.name}"
+  create_policy         = var.role_policy != null ? true : false
+  principal_type        = "Service"
```
Comment (skesarkar-schubergphilis) [inline iam.tf:7]: policy_arns = ["arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"]

File: iam.tf [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  principal_identifiers = ["glue.amazonaws.com"]
+  role_policy           = var.role_policy
+  tags                  = var.tags
+}
+
+resource "aws_iam_role_policy_attachment" "job_execution_role" {
```
Comment (skesarkar-schubergphilis) [inline iam.tf]: We can specify the policy arn within the role resource itself so this can be removed

File: iam.tf (hunk 1 above)
Comment (thulasirajkomminar) [reply iam.tf:7]: fixed it

File: iam.tf (hunk 2 above)
Comment (thulasirajkomminar) [reply iam.tf]: fixed it

**Review by skesarkar-schubergphilis — APPROVED** [review]

**Pull request merged**